tests/scripts/all.sh

```diff
@@ -4005,14 +4009,45 @@ test_build_opt () {
     done
 }
 
-component_test_clang_opt () {
+# For FreeBSD we invoke the function by name so this condition is added
+# to disable the existing test_clang_opt function for linux.
+if [[ $(uname) != "Linux" ]]; then
+    component_test_clang_opt () {
+        scripts/config.py full
+        test_build_opt 'full config' clang -O0 -Os -O2
+    }
+fi
+
+component_test_clang_latest_opt () {
+    scripts/config.py full
+    test_build_opt 'full config' clang-latest -O0 -Os -O2
```

Comment:
Instead of running `clang-latest` and such, please run `"$CLANG_LATEST"` and such, and set them to default to their current names but allow changing them with command line options `--clang-latest` and such. The same thing we're doing with `--openssl-next` setting `OPENSSL_NEXT`, etc. And with the same environment overrides, while we're at it. (Which means that we don't actually need the symlinks on the CI, but we can handle that later.)

Reply:
Once this PR is merged, I will create a new PR on `mbedtls-test` removing the symlinks.

Reply:
More precisely: setting environment variables instead of creating symlinks.

Reply:
I'm planning to set `"$CLANG_LATEST"` to `clang-16` as the default value within `all.sh`. So do we still need to set environment variables ?